### PR TITLE
Add Tesla Middleware To Log Requests And Responses

### PR DESCRIPTION
**Pull request by nvieirafelipe**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,8 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+### Added
+- Add tesla middleware to log requests and responses.
```
Comment (prodis) [inline CHANGELOG.md:9]:
```suggestion
- Add Tesla middleware to log requests and responses.
```

Missing to mention the changes in the CI.

File: lib/tesla/middleware/meta_logger.ex [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+  @spec maybe_put_default_value(Env.opts(), atom(), any()) :: Env.opts()
+  defp maybe_put_default_value(options, key, default_value),
+    do: Keyword.put(options, key, Keyword.get(options, key, default_value))
+
+  @spec log_request(Env.t(), Env.opts()) :: Env.t()
+  defp log_request(%Env{body: body, method: method, query: query, url: url} = env, options) do
+    level = Keyword.get(options, :log_level)
+    headers = build_headers(env, options)
+
+    formatted_method =
```
Comment (prodis) [inline lib/tesla/middleware/meta_logger.ex:63]: Move to a separate function?

**Review by prodis — APPROVED** [review]
LGTM

File: lib/tesla/middleware/meta_logger.ex [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+  defp response_log_level({:ok, %Env{status: status}}, _options) when status >= 300, do: :warn
+  defp response_log_level(_result, options), do: Keyword.get(options, :log_level)
+
+  @spec filter_header({String.t(), String.t()}, [String.t()]) :: {String.t(), String.t()}
+  defp filter_header({key, _value} = header, filter_headers),
+    do: if(key in filter_headers, do: {key, "[FILTERED]"}, else: header)
+
+  @spec format_method(atom()) :: String.t()
+  defp format_method(method) do
+    formatted_method =
```
Comment (sergioaugrod) [inline lib/tesla/middleware/meta_logger.ex:106]:
```suggestion
```

**Review by sergioaugrod — APPROVED** [review]

**Review by zekus — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -1,6 +1,6 @@
 # MetaLogger
 
-[![Build Status](https://travis-ci.com/FindHotel/meta_logger.svg?branch=master)](https://travis-ci.com/FindHotel/meta_logger)
+![meta_logger](https://github.com/FindHotel/meta_logger/workflows/meta_logger/badge.svg?branch=master)
 
 Wrapper for Elixir.Logger that keeps logger metadata from caller processes.
```
Comment (prodis) [inline README.md:5]: Please add an usage example of the Tesla middleware. Or at least just mention that it exists.

**Pull request merged**